tools/c7n_azure/c7n_azure/resources/key_vault.py

```diff
@@ -114,3 +121,89 @@ def enhance_policies(self, access_policies):
             policy['principalName'] = GraphHelper.get_principal_name(aad_object)
 
         return access_policies
+
+
+@resources.register('keyvault-key')
+class KeyVaultKey(ArmResourceManager):
+
+    class resource_type(ArmResourceManager.resource_type):
+        service = 'azure.mgmt.keyvault'
+        client = 'KeyVaultManagementClient'
+        enum_spec = ('vaults', 'list', None)
+
+    def augment(self, resources):
+        s = Session(resource='https://' + RESOURCE_KEYVAULT_BASEURL)
+        self.keyvault_client = KeyVaultClient(s.get_credentials())
+        r = []
+
+        for ritem in resources:
+
+            vault_url = 'https://' + ritem["name"] + "." + RESOURCE_KEYVAULT_BASEURL
+
+            vkeys = self.keyvault_client.get_keys(vault_url)
+
+            try:
+                for vkey in vkeys:
+                    keyname = KeyVaultId.parse_key_id(vkey.kid).name
+                    r.append({'kid': vkey.kid,
+                              'keyname': keyname,
+                              'keyvault': {'id': ritem["id"],
+                                           'name': ritem["name"],
+                                           'location': ritem["location"],
+                                           'tags': ritem["tags"]},
+                              'enabled': vkey.attributes.enabled,
+                              'created': vkey.attributes.created,
+                              'updated': vkey.attributes.updated,
+                              'tags': vkey.tags})
+            except KeyVaultErrorException as e:
```

Comment:
there isn't any network call that would generate this exception in the try/except block. parse_key_id is a client local string parse that raises valueerror as well.

https://github.com/Azure/azure-sdk-for-python/blob/master/azure-keyvault/azure/keyvault/key_vault_id.py#L270